src/Activity.js

```diff
@@ -65,23 +76,25 @@ class Activity {
       return `${userName}: ${listItem[userID]}`
     })
   }
+
   showcaseWinner(user, date, userRepo) {
-    let namedList = this.showChallengeListAndWinner(user, date, userRepo);
-    let winner = this.showChallengeListAndWinner(user, date, userRepo).shift();
-    return winner;
+    return this.showChallengeListAndWinner(user, date, userRepo)[0];
   }
+
   getStreak(userRepo, id, relevantData) {
     let data = this.activityData;
     let sortedUserArray = (userRepo.makeSortedUserArray(id, data)).reverse();
-    let streaks = sortedUserArray.filter(function(element, index) {
-      if (index >= 2) {
-        return (sortedUserArray[index - 2][relevantData] < sortedUserArray[index - 1][relevantData] && sortedUserArray[index - 1][relevantData] < sortedUserArray[index][relevantData])
+    let streaks = sortedUserArray.reduce((acc, userData, index, arr) => {
```

Comment:
Can these be consts or is let the best way to go?

Reply:
NOICE REDUCE!

Reply:
done